File: job-offer/src/main/java/com/oxygensend/joboffer/domain/repository/AddressRepository.java
package com.oxygensend.joboffer.domain.repository;

import com.oxygensend.joboffer.domain.entity.Address;
import java.util.List;
import java.util.Optional;

public interface AddressRepository {

    Optional<Address> findByPostCodeAndCity(String postCode, String city);
    long count();
    void saveAll(Iterable<Address> addresses);

    List<Address> findAll();
}

File: static-data/src/main/kotlin/com/oxygensend/staticdata/domain/AddressRepository.kt
package com.oxygensend.staticdata.domain

interface AddressRepository {

    fun findAll(search: String?): List<Address>
    fun findAll(): List<Address>
    fun save(address: Address): Address
    fun findByCity(city: String): Address?
    fun saveBatch(addresses: List<Address>)
    fun count(): Long
    fun saveAll(addresses: List<Address>)
}